pkgs/applications/science/math/sage/sage-src.nix

```diff
@@ -78,6 +78,20 @@ stdenv.mkDerivation rec {
 
     # ignore a deprecation warning for usage of `cmp` in the attrs library in the doctests
     ./patches/ignore-cmp-deprecation.patch
+
+    # https://trac.sagemath.org/ticket/30801. this patch has not been
+    # merged upstream yet, so we don't use fetchSageDiff because it
+    # returns a file that contains each commit as a separate patch
+    # instead of a single diff, and some commits from the pari update
+    # branch are already in 9.3.rc5 (auto-resolvable merge conflicts).
+    #
+    # TODO: this patch is from 2021-04-21, check for changes once the
+    # upstream ticket is merged.
+    (fetchpatch {
+      name = "pari-2.13.1.patch";
+      url = "https://github.com/sagemath/sagetrac-mirror/compare/c27d4d6803d04a0f25377caa59de530060529c7a...558b5e820c169b616477f8a2e0eac47f07e41b91.diff";
```

Comment:
I think all patches, once merged to Sage main, have `origin/develop` as their immediate parent, and in the process of merging such conflicts get resovled. Therefore, the problem described above only affects unmerged Trac branches. `sagemath/sage` does not contain trac branches, so we wouldn't see any change in using GitHub unless we also migrated to `sagemath/sagetrac-mirror`. But using `sagemath/sagetrac-mirror` would be impossible because the tags aren't synchronized properly there (the most recent tag now is `9.3beta2`).

Let me know if I should still switch to GitHub given that, I might be missing something.

Reply:
To be clear, what GitHub gives us is the ability of choosing between `compare/commit1..commit2.patch`, which returns a "non-squashed" version of the commits that happened from `common_ancestor(commit1, commit2)` to `commit2` and `compare/commit..commit2.diff`, which is just a diff between the two branches. `fetchSageDiff` only offers the first possibility.

Reply:
> But using `sagemath/sagetrac-mirror` would be undesirable because the tags aren't synchronized properly there (the most recent tag now is `9.3beta2`).

You are right, that would be inconvenient. I didn't know that. Thanks for clarifying.